lib/Doctrine/ORM/Persisters/Entity/BasicEntityPersister.php

```diff
@@ -1580,11 +1581,22 @@ public function lock(array $criteria, $lockMode)
      */
     protected function getLockTablesSql($lockMode)
     {
+        if ($lockMode === null) {
+            Deprecation::trigger(
+                'doctrine/orm',
+                'https://github.com/doctrine/orm/pull/9466',
+                'Passing null as argument to %s is deprecated, pass LockMode::NONE instead.',
+                __METHOD__
+            );
+
+            $lockMode = LockMode::NONE;
+        }
```

Comment:
In this method, `$lockMode === null` actually means `LockMode::NONE`. Disallowing `null` here makes the calls more explicit and allows us to narrow down the type to `int`.

I don't know if that method is ever called from outside this repository, so I'm not sure if it's worth mentioning this deprecation in the upgrade log.

Reply:
I guess everything that is open for extension can result in a BC break.